test/utils/padder/padder.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package padder
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nrtv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	"github.com/openshift-kni/numaresources-operator/test/utils/fixture"
+	nrtutil "github.com/openshift-kni/numaresources-operator/test/utils/noderesourcetopologies"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects/wait"
+)
+
+// This package allows to control the amount of available allocationTarget under the nodes.
+// it's doing that by "padding" the nodes with guaranteed pods until node reaches the desired amount of available allocationTarget.
+//
+// Usage example:
+// pad := New(client, "test-ns")
+// pad.Nodes(2).UntilAvailableIsResource(corev1.ResourceCPU, "3").UntilAvailableIsResource(corev1.ResourceMemory, "100Mi").Pad()
+// or
+// rl := corev1.ResourceList {
+//		corev1.ResourceCPU: "10",
+//		corev1.ResourceMemory: "100Mi"
+//	}
+// pad.Nodes(2).UntilAvailableIsResourceList(rl).Pad()
+
+const PadderLabel = "nrop-test-pad-pod"
+
+type Padder struct {
+	// Client defines the API client to run CRUD operations, that will be used for testing
+	Client    client.Client
+	namespace string
+	*padRequest
+}
+
+type padRequest struct {
+	nNodes           int
+	allocationTarget corev1.ResourceList
+}
+
+// New return new padder object
+// ns is where padding pods will get deployed
+func New(cli client.Client, ns string) (*Padder, error) {
+	return &Padder{
+		Client:     cli,
+		namespace:  ns,
+		padRequest: &padRequest{allocationTarget: make(corev1.ResourceList)},
+	}, nil
+}
+
+// Nodes number of nodes to pad
+func (p *Padder) Nodes(n int) *Padder {
+	p.padRequest.nNodes = n
```

Comment:
this is suspicious. If we need to restrict the nodes to pad from the set we autodetect, we would need to restrict using a stronger criteria (node name? labels?). If this is just to signal how many nodes we are expected to pad, we can easily autodetect.

Reply:
we just need to know how many nodes are going to be padded we don't actually care from their names/label.
let me explain using an exmaple:
I have 4 nodes (Worker nodes with single-numa-node) I'm padding 3 out 4 nodes until available CPUs reaches 3.
Then I create a test pod which asks for 4 CPUs. 
And I know I only have one worker nodes which can handle that requests (because the 3 others have only 3 available CPUs)